app/javascript/src/utils/compiler/variables.js

```diff
@@ -138,6 +138,7 @@ function getLiteralPlayerVariables(source, stringRanges) {
 }
 
 export function getVariables(joinedItems) {
+  joinedItems = evaluateParameterObjects(joinedItems)
```

Comment:
I think we should add checks for this in variables.test.js too

Reply:
Yep, and you can run them locally with
```bash
npx jest
```

Or there are also a few VSCode extensions that add UI to run them, like [firsttris.vscode-jest-runner](https://marketplace.visualstudio.com/items?itemName=firsttris.vscode-jest-runner)